lib/galaxy/webapps/galaxy/services/history_contents.py

```diff
@@ -1510,6 +1513,10 @@ def _change_dbkey(self, item: HistoryItemModel, params: ChangeDbkeyOperationPara
 
     def _add_tags(self, trans: ProvidesUserContext, item: HistoryItemModel, params: TagOperationParams):
         trans.tag_handler.add_tags_from_list(trans.user, item, params.tags, flush=self.flush)
+        # Changing tags does not change the item, but we need to update the history to trigger the completion of the operation
+        item.update()
 
     def _remove_tags(self, trans: ProvidesUserContext, item: HistoryItemModel, params: TagOperationParams):
         trans.tag_handler.remove_tags_from_list(trans.user, item, params.tags, flush=self.flush)
+        # Changing tags does not change the item, but we need to update the history to trigger the completion of the operation
+        item.update()
```

Comment:
Should those updates instead be moved to the tag handler ? I assume we always want the update time to be altered ?

Reply:
That is what I had in the previous attempt but many tests failed with a 500 status code... I can give it another try and see what it is actually upsetting those tests

Reply:
I think the issue with this is that some of the database models that may call `set_tags_from_list` did not inherit from `UsesCreateAndUpdateTime` like `StoredWorkflow`. `Page`, etc. Does it make more sense to add `UsesCreateAndUpdateTime` to these models instead of skipping the call in https://github.com/galaxyproject/galaxy/pull/18068/commits/983f7c37c7a96dae63ed6271963d3261666a954b?

Reply:
I don't see why not. As long as they have those two fields and don't have their own `update` method that does something different, then sure.

Reply:
Great! I've done it in https://github.com/galaxyproject/galaxy/pull/18068/commits/51d8743b5f174abb41bffe3259c50a0b19212aee it feels more consistent to me now. Adding or removing tags in most items (including Histories) was not triggering an update of the item.update_time in the past but it feels like it should as Marius suggested.